src/HomogeneousCalorimeter_geo.cpp

```diff
@@ -406,58 +406,114 @@ static std::tuple<int, int> add_disk(Detector& desc, Assembly& env, xml::Collect
+
+
+  // Version3: solid with elliptical inside
   //
-  Tube   Ssolid_ring12(8.5 * cm, rmin, calo_module_length / 2., phimin, phimax);
-  Volume Sring12_vol("Sring12", Ssolid_ring12, ring_material);
-  Sring12_vol.setVisAttributes(desc.visAttributes(plm.attr<std::string>(_Unicode(vis_struc))));
-  Transform3D tr_global_Iring = RotationZYX(0., 0., 0.) * Translation3D(0., 0., 0.);
-
-  // Supporting frame for the cabling
-  // Having overlapped with tracker barrelendcapsupporting structure, comment here until the size and position
-  // determined
-  //
+  Box inner_support_main(8.2*cm, 6.15*cm, calo_module_length/2.);  // Original size
+  // Box inner_support_main(8.2*cm, 5.125*cm, calo_module_length/2.);  // Adapted size
+  Box subtract_corner(1.025*cm, 1.025*cm, calo_module_length/2.);
+  EllipticalTube   subtract_a(7.5*cm, 5.5*cm, calo_module_length/2.);
+  SubtractionSolid inner_support_substractb1(inner_support_main, subtract_corner, Position(7.175*cm, 5.125*cm, 0.));
+  SubtractionSolid inner_support_substractb2(inner_support_substractb1, subtract_corner, Position(7.175*cm, -5.125*cm, 0.));
+  SubtractionSolid inner_support_substractb3(inner_support_substractb2, subtract_corner, Position(-7.175*cm, 5.125*cm, 0.));
+  SubtractionSolid inner_support_substractb4(inner_support_substractb3, subtract_corner, Position(-7.175*cm, -5.125*cm, 0.));
+  SubtractionSolid inner_support_substracta(inner_support_substractb4, subtract_a, Position(0., 0., 0.));
+  Volume           inner_support_vol("inner_support_vol", inner_support_substracta, inner_ring_material);
+  inner_support_vol.setVisAttributes(desc.visAttributes(plm.attr<std::string>(_Unicode(vis_struc))));
+  Transform3D tr_global_Iring_elli = RotationZYX(-15.*degree, 0., 0.) * Translation3D(0., 0., 0.);
+
+
+  // // The vacuum inside the inner structure
+  // //
+  // EllipticalTube   inner_elliptical_vacuum(7.5*cm, 4.5*cm, calo_module_length/2.);
+  // Volume           inner_elliptical_vacuum_vol("inner_elliptical_vacuum_vol", inner_elliptical_vacuum, hole_material);
+  // inner_elliptical_vacuum_vol.setVisAttributes(desc.visAttributes(plm.attr<std::string>(_Unicode(vis_struc))));
+  // Transform3D tr_global_Iring_elli_vacuum = RotationZYX(-15.*degree, 0., 0.) * Translation3D(1.025*cm, 0., 0.);
+
+
+  //=============================
+  // The mother volume of modules
+  //=============================
+  bool has_envelope = dd4hep::getAttrOrDefault<bool>(plm, _Unicode(envelope), false);
+  PolyhedraRegular solid_world(12, 0., r12min, calo_module_length);
+  EllipticalTube  solid_sub(7.5*cm, 5.5*cm, calo_module_length/2.);
+  Transform3D subtract_pos = RotationZYX(-15.*degree, 0., 0.) * Translation3D(0., 0., 0.);
+  SubtractionSolid calo_subtract(solid_world, solid_sub, subtract_pos);
+  Volume      env_vol(std::string(env.name()) + "_envelope", calo_subtract, outer_ring_material);
+  Transform3D tr_global = RotationZYX(15.*degree, 0., 0.) * Translation3D(0., 0., 0.);
```

Comment:
This should not use hardcoded constants like 7.5 cm, 5.5 cm, 15 degrees. The motivation of the compact files is that they contain the parameters, and the plugin contains what is done with those parameters.

Reply:
Remove the those number and put them into the configuration files